### DISCUSSION TITLE
add_stream_from_template sucks, and also fails when using input_stream that is mp3

### DISCUSSION THREAD

**Discussion by nirdeepdub** [PyAV-Org/PyAV#1670] [3. General]
I don't see any added value from this new interface to replace the original, why not add a call to the function if "template=" is sent to add_stream Instead of expecting us to modify our code.

In addition it would be lovely to get a dict representation of all codec defining params. so that they'll be 

## Example API
```
j = input_stream.codec.dict()
j['layout'] = new layout.
output_stream.add_stream(**j)
```

## Additional context
https://github.com/deepdub-ai/audiosample

**Comment by WyattBlue** [comment]
`add_stream_from_template` can be fully typed, unlike the original API.
The `mp3` codec is fixed in 14.0.1

**Reply by nirdeepdub** [reply]
Great news about the fix.
What about the ability to extract all of the params that can then be used to re-enter as the example above ? That would be a very friendly interface.

**Reply by WyattBlue** [reply]
PRs are welcome
